pager.c

```diff
@@ -310,14 +308,21 @@ inline static bool assert_mailbox_writable(struct Mailbox *mailbox)
     break;                                                                     \
   }
 
-#define CHECK_ACL(aclbit, action)                                              \
-  if (!Contex2 || !(Contex2->mailbox->rights & aclbit))                        \
-  {                                                                            \
-    mutt_flushinp();                                                           \
-    /* L10N: %s is one of the CHECK_ACL entries below. */                      \
-    mutt_error(_("%s: Operation not permitted by ACL"), action);               \
-    break;                                                                     \
+inline static bool assert_mailbox_permissions(struct Mailbox *m, AclFlags acl, char* action)
+{
+  assert(m);
+  if (m->rights & acl) 
+  {
+      return true;
+  }
+  else
```

Comment:
We can drop the `else` here (slightly flattening the code).
If the `if` statement is true, the function ends.

Reply:
done